src/data_collector/indicator_pipeline/indicator_pipeline.py

```
@@ -440,88 +616,44 @@
+                            "feature_name": feature_name,
+                            "feature_value": feature_value,
+                            "quality_score": quality_score_val,
+                        }
+                    )
 
-                        saved_count += 1
-
-                conn.commit()
-
+            saved_count = bulk_upsert_technical_features(rows, page_size=1000, overwrite=overwrite)
             return saved_count
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)